src/auction/utils.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable camelcase, compat/compat */
+
+/*
+ * @function
+ * @name getEligible
+ * @description filter all buyers by their owner field to ensure they're eligible to bid
+ * @author Newton <cnewton@magnite.com>
+ * @param {array<Object>} buyers - an array of objects containing interest group buyers
+ * @param {array<String>} eligible - a list of eligible owners to check against
+ * @return {Array<Object> | null} an array of objects; null if none found;
+ */
+export const getEligible = (groups, eligibility) => {
+	if (eligibility === '*') {
+		return groups;
+	}
+
+	const eligible = groups.filter(({ owner }) => eligibility.includes(owner));
+	if (eligible.length) {
+		return eligible;
+	}
+
+	return null;
+};
+
+/*
+ * @function
+ * @name getBids
+ * @description given a set of bidders, grab their bid
+ * @author Newton <cnewton@magnite.com>
+ * @param {array<Object>} bidders - a list of eligible owners to bid
+ * @param {array<Object>} conf - an auction configuration object
+ * @return {object | null} an array of objects containing bids; null if none found
+ */
+export const getBids = async (bidders, conf) => Promise.all(
+	bidders.map(async bidder => {
+		const { generate_bid } = await import(bidder.bidding_logic_url);
+
+		// check if there is even a generate_bid function
+		// if not, removed bidder from elibility
+		if (!generate_bid && typeof generate_bid !== 'function') {
+			return null;
+		}
+
+		// @TODO: need to figure out how to pull in trusted bidding signals
+		const bid = generate_bid(bidder, conf?.auction_signals, conf?.per_buyer_signals?.[bidder.owner], conf?.trusted_bidding_signals, {
+			top_window_hostname: window.top.location.hostname,
+			seller: conf.seller,
+		});
+
+		// check if generate_bid function returned the necessary parts to score
+		// if not, removed bidder from elibility
+		if (!(
+			(bid.ad && typeof bid.ad === 'object') &&
+			(bid.bid && typeof bid.bid === 'number') &&
+			(bid.render && (typeof bid.render === 'string' || Array.isArray(bid.render)))
+		)) {
+			return null;
+		}
+
+		return {
+			...bidder,
```

Comment:
```suggestion
			... interestGroup,
```

Reply:
at this time, going to go with bidders; if the general consensus is to change it, will do so at that time